tests/unit/functions.tcl

```diff
@@ -1132,4 +1164,38 @@ start_server {tags {"scripting"}} {
         r function flush
         r function stats
     } {running_script {} engines {LUA {libraries_count 0 functions_count 0}}}
+
+    test {FUNCTION - function test empty engine} {
+         catch {r function load replace {#! name=test
+            redis.register_function('foo', function() return 1 end)
+        }} e
+        set _ $e
+    } {ERR Engine not found}
+
+    test {FUNCTION - function test unknown metadata value} {
+         catch {r function load replace {#!lua name=test foo=bar
+            redis.register_function('foo', function() return 1 end)
+        }} e
+        set _ $e
+    } {ERR Invalid metadata value given: foo=bar}
```

Comment:
maybe we rather ignore unrecognized metadata (forward compatibility)?
i see we didn't do that in the EVAL shebang, so i suppose we already discussed and decided that.

Reply:
But then if we will add version for example, the user will think that the version was accepted while actually is was ignored, not sure what is better ..